src/openrct2/paint/tile_element/Paint.Surface.cpp

```diff
@@ -1279,8 +1279,8 @@ void surface_paint(paint_session* session, uint8_t direction, uint16_t height, c
 
     if (!(session->ViewFlags & VIEWPORT_FLAG_HIDE_VERTICAL))
     {
-        const uint32_t edgeStyle = tileElement->AsSurface()->GetEdgeStyle();
-        if (edgeStyle >= TERRAIN_EDGE_COUNT)
+        const auto edgeStyle = tileElement->AsSurface()->GetEdgeStyle();
+        if (static_cast<int32_t>(edgeStyle) >= object_entry_group_counts[EnumValue(ObjectType::TerrainEdge)])
```

Comment:
Should we use `EnumValue` for `edgeStyle` instead?

Reply:
???

Reply:
Yeah I mean instead of the static_cast using our utility function called `EnumValue` or something like it

Reply:
edgeStyle is not an enum value (not anymore), so there is no reason to do that.